subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -598,6 +673,73 @@ static void hawkbit_dump_deployment(struct hawkbit_dep_res *d)
 	LOG_DBG("md5sum =%s", l->md5sum_http.href);
 }
 
+#ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
+int hawkbit_set_server_addr(char *addr_str)
+{
+	if (!strcmp(addr_str, hb_cfg.server_addr)) {
+		LOG_DBG("hawkbit server address already correctly configured");
+	} else if (k_sem_take(&probe_sem, HAWKBIT_SET_SERVER_TIMEOUT) == 0) {
+		strncpy(hb_cfg.server_addr, addr_str, sizeof(hb_cfg.server_addr));
+		LOG_DBG("configured hawkbit server address: %s", hb_cfg.server_addr);
+		k_sem_give(&probe_sem);
+	} else {
+		LOG_WRN("failed setting hawkbit server address");
+		return -EAGAIN;
+	}
+	return 0;
+}
+
+int hawkbit_set_server_port(uint16_t port)
+{
+	if (atoi(hb_cfg.server_port) == port) {
+		LOG_DBG("hawkbit server port already correctly configured");
+	} else if (k_sem_take(&probe_sem, HAWKBIT_SET_SERVER_TIMEOUT) == 0) {
+		snprintf(hb_cfg.server_port, sizeof(hb_cfg.server_port), "%u", port);
+		LOG_DBG("configured hawkbit server port: %s", hb_cfg.server_port);
```

Comment:
this and others log messages: can potentially reuse the same `"hawkbit/server_port"` key used in the settings:

```c
LOG_DBG("configured %s: %s", "hawkbit/server_port", hb_cfg.server_port);
```